app/unit-tests/src/org/commcare/android/tests/processing/QuarantinedFormsTest.java

```diff
@@ -0,0 +1,31 @@
+package org.commcare.android.tests.processing;
+
+import org.commcare.utils.FormUploadUtil;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+
+/**
+ * Created by amstone326 on 6/14/17.
+ */
+
+public class QuarantinedFormsTest {
```

Comment:
@ctsims That's essentially what I tried to do (if you look at the full commit history of this PR you'll see that I iterated on that for a while and then ultimately removed it). My decision was based on the fact that a) The test was failing in a way that I couldn't figure out the cause of after trying for quite a while, and b) That I was mocking a large portion of the behavior, which made the usefulness of the test feel really limited. Based on those 2 things, I decided a calabash test would be both easier (or at least take less of my time) and a "better" test. 

On the flip side, the challenge of writing a calabash test for this is having a form submission that will consistently trigger this response from HQ. In order to determine the feasibility of that, I would need to talk with Ben or whoever works on this for HQ to determine what situations will trigger this response, and then think about which (if any) is something that we can reliably recreate on mobile. So I think it makes sense to wait on deciding about this until I can have that converesation. If reproducing in a calabash test is feasible, I do think I prefer that, for the reasons I outlined above.